admin/class-import-wpseo.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * @package WPSEO\Admin\Import\External
+ */
+
+/**
+ * Class WPSEO_Import_WPSEO
+ *
+ * Class with functionality to import Yoast SEO settings from wpSEO
+ */
+class WPSEO_Import_WPSEO extends WPSEO_Import_External {
+
+	/**
+	 * @var array
+	 */
+	private $robot_values = array(
+		1 => array( 'index' => 2, 'follow' => 0 ),
+		2 => array( 'index' => 2, 'follow' => 1 ),
+		3 => array( 'index' => 1, 'follow' => 0 ),
+		4 => array( 'index' => 1, 'follow' => 0 ),
+		5 => array( 'index' => 1, 'follow' => 1 ),
+		6 => array( 'index' => 2, 'follow' => 0 ),
+	);
+
+
+	/**
+	 * Import All In One SEO settings
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		$this->import_post_metas();
+		$this->import_taxonomy_metas();
+
+		$this->set_msg( __( 'wpSEO data successfully imported. ', 'wordpress-seo' ) );
+	}
+
+	/**
+	 * Import the post meta values
+	 */
+	private function import_post_metas() {
+		WPSEO_Meta::replace_meta( '_wpseo_edit_title', WPSEO_Meta::$meta_prefix . 'title', $this->replace );
+		WPSEO_Meta::replace_meta( '_wpseo_edit_description', WPSEO_Meta::$meta_prefix . 'metadesc', $this->replace );
+		WPSEO_Meta::replace_meta( '_wpseo_edit_description', WPSEO_Meta::$meta_prefix . 'keywords', $this->replace );
+		WPSEO_Meta::replace_meta( '_wpseo_edit_canonical', WPSEO_Meta::$meta_prefix . 'canonical', $this->replace );
+
+		$this->import_post_robots();
+
+	}
+
+	/**
+	 * Importing the robot values from WPSEO plugin. These have to be converted
+	 */
+	private function import_post_robots() {
+		$query_posts  = new WP_Query( 'post_type=any&meta_key=_wpseo_edit_robots&order=ASC' );
```

Comment:
Perhaps you can split up the saving and deleting the meta values to new functions so this function will only be responsible for getting the post meta to improve single responsibility.